### PR TITLE
Shortest Paths Module Update

### PR DESCRIPTION
**Pull request by nwatx**

### CODE REVIEW

**Review by ryanchou-dev — APPROVED** [review]
👍

File: content/4_Gold/Shortest_Paths.mdx [hunk 1]
```diff
@@ -66,6 +64,111 @@ idea to learn Bellman-Ford and Floyd-Warshall first since they're simpler.
...
+	cin >> n >> m >> q;
+
+	vector<vector<ll>> mat(n, vector<ll>(n, INF));
+
+	for (int i = 0; i < m; i++) {
+		int u, v, w;
+		cin >> u >> v >> w, u--, v--;
+		if (w < mat[u][v]) mat[u][v] = mat[v][u] = w;
+	}
+
```
Comment (SansPapyrus683) [inline content/4_Gold/Shortest_Paths.mdx:102]: just add a comment like "floyd-warshall"

File: content/4_Gold/Shortest_Paths.mdx [hunk 2]
```diff
@@ -66,6 +64,111 @@ idea to learn Bellman-Ford and Floyd-Warshall first since they're simpler.
...
+		for(long[] row : mat) {
+			Arrays.fill(row, INF);
+		}
+
+		for (int i = 0; i < m; i++) {
+			int u = io.nextInt(), v = io.nextInt(), w = io.nextInt();
+			u--; v--;
+			if (w < mat[u][v]) mat[u][v] = mat[v][u] = w;
+		}
```
Comment (SansPapyrus683) [inline content/4_Gold/Shortest_Paths.mdx:147]: same for java code

File: content/4_Gold/Shortest_Paths.mdx [hunk 3]
```diff
@@ -66,6 +64,111 @@ idea to learn Bellman-Ford and Floyd-Warshall first since they're simpler.
...
+
+		for (int k = 0; k < n; k++)
+			for (int i = 0; i < n; i++)
+				for (int j = i + 1; j < n; j++)
+					if (mat[i][k] + mat[k][j] < mat[i][j])
+						mat[i][j] = mat[j][i] = mat[i][k] + mat[k][j];
+
+		while (q-- > 0) {
```
Comment (SansPapyrus683) [inline content/4_Gold/Shortest_Paths.mdx:155]: why not just be like everyone else and use a for loop lol

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/4_Gold/Shortest_Paths.mdx [hunk 4]
```diff
@@ -66,6 +64,114 @@ idea to learn Bellman-Ford and Floyd-Warshall first since they're simpler.
...
+			Arrays.fill(row, INF);
+		}
+
+		for (int i = 0; i < m; i++) {
+			int u = io.nextInt(), v = io.nextInt(), w = io.nextInt();
+			u--; v--;
+			if (w < mat[u][v]) mat[u][v] = mat[v][u] = w;
+		}
+
+
```
Comment (SansPapyrus683) [inline content/4_Gold/Shortest_Paths.mdx:150]:
```suggestion
```

**Review by SansPapyrus683 — APPROVED** [review]
pppplease use for loops though

File: content/4_Gold/Shortest_Paths.mdx [hunk 5]
```diff
@@ -66,6 +64,113 @@ idea to learn Bellman-Ford and Floyd-Warshall first since they're simpler.
...
+public class Main {
+	// CodeSnip{Kattio}
+
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+
+		int n = io.nextInt(), m = io.nextInt(), q = io.nextInt();
+
+		long[][] mat = new long[n][n];
+		long INF = (long)1e18;
```
Comment (nathangong) [inline content/4_Gold/Shortest_Paths.mdx:138]: Why not just Long.MAX_VALUE

**Review by nathangong — APPROVED** [review]

File: content/4_Gold/Shortest_Paths.mdx (hunk 5 above)
Comment (HarryWangATX) [reply content/4_Gold/Shortest_Paths.mdx:138]: its good practice to set inf as a little below the max value so you dont overflow.

Comment (nathangong) [reply content/4_Gold/Shortest_Paths.mdx:138]: wouldn't you be doing something wrong if you're adding to INF anyways?

**Pull request merged**